wit/types.wit

```diff
@@ -159,20 +159,18 @@ interface types {
     /// list with the same key.
     ///
     /// The tuple is a pair of the field key, represented as a string, and
-    /// Value, represented as a list of bytes. In a valid Fields, all keys
-    /// and values are valid UTF-8 strings. However, values are not always
-    /// well-formed, so they are represented as a raw list of bytes.
+    /// Value, represented as a list of bytes.
     ///
-    /// An error result will be returned if any header or value was
-    /// syntactically invalid, or if a header was forbidden.
+    /// An error result will be returned if any `field-key` or `field-value` is
+    /// syntactically invalid, or if a field is forbidden.
     from-list: static func(
       entries: list<tuple<field-key,field-value>>
     ) -> result<fields, header-error>;
 
     /// Get all of the values corresponding to a key. If the key is not present
-    /// in this `fields`, an empty list is returned. However, if the key is
-    /// present but empty, this is represented by a list with one or more
-    /// empty field-values present.
+    /// in this `fields` or is syntactically invalid, an empty list is returned.
+    /// However, if the key is present but empty, this is represented by a list
+    /// with one or more empty field-values present.
```

Comment:
This seems like the right choice given that 0.2.0 is frozen and we can't change the types.  But the next time we can make a breaking change (e.g., 0.3.0), I wonder if a syntactically-invalid key parameter should produce an error and thus the return value should be wrapped in a `result`?  The answer doesn't affect this PR, but separately we could start a list of "pending tweaks for 0.3.0" if we agree.

Reply:
I think that would be more consistent with the other methods, and would support carrying it forward as a proposed change for 0.3.0.